### ISSUE TITLE
ERROR: type String has no field head

### ISSUE THREAD

**Issue by innerlee** [tonyhffong/Lint.jl#167]
```julia
julia> using Lint

julia> lintstr("""type love
       who
       "Julia :)"
       love() = "yes"
       end
       """)
ERROR: type String has no field head
 in linttype(::Expr, ::Lint.LintContext) at C:\Users\lizhi\.julia\v0.5\Lint\src\types.jl:99
 in lintexpr(::Expr, ::Lint.LintContext) at C:\Users\lizhi\.julia\v0.5\Lint\src\Lint.jl:228
 in lintstr(::String, ::Lint.LintContext, ::Int64) at C:\Users\lizhi\.julia\v0.5\Lint\src\Lint.jl:151
 in lintstr(::String) at C:\Users\lizhi\.julia\v0.5\Lint\src\Lint.jl:124
```

This error happens if you want to add some inline doc to the inner constructor. One way to bypass this error:
at https://github.com/tonyhffong/Lint.jl/blob/master/src/types.jl#L99,
before the first use of ``def.head``, add the following code
```julia
elseif !(:head in fieldnames(def))
    nothing
```

**Comment by TotalVerb** [comment]
Does this documentation syntax actually work?

**Comment by innerlee** [comment]
Oh, that's a good question... 😕 Maybe a good linter message instead of error?

**Comment by TotalVerb** [comment]
Yes, I think that's the better approach. As far as I can tell, documenting outer constructors works, while documenting inner constructors doesn't seem to do anything.

**Comment by galenlynch** [comment]
I am having a similar problem with valid documentation string syntax. Type definitions with field documentation seems to cause the same error reported by @TotalVerb. I first encountered this with the following code:

```julia
"""
    ContinuousFile(io::IOStream)
Type for an open continuous file.

# Fields

**`io`** `IOStream` object.

**`nsample`** number of samples in a file.

**`nblock`** number of data blocks in a file.

**`header`** [`OriginalHeader`](@ref) of the current file.
"""
immutable ContinuousFile{T<:Integer, S<:Integer, H<:OriginalHeader}
    "IOStream for open continuous file"
    io::IOStream
    "Number of samples in file"
    nsample::T
    "Number of data blocks in file"
    nblock::S
    "File header"
    header::H
end
```

But it can also be recreated with simpler REPL examples, e.g.:
```julia
julia> lintstr("""
       immutable foo
       "bar field"
       bar::Int
       end
       """)
ERROR: type String has no field head
 in linttype(::Expr, ::Lint.LintContext) at /home/galen/.julia/v0.5/Lint/src/types.jl:99
 in lintexpr(::Expr, ::Lint.LintContext) at /home/galen/.julia/v0.5/Lint/src/Lint.jl:228
 in lintstr(::String, ::Lint.LintContext, ::Int64) at /home/galen/.julia/v0.5/Lint/src/Lint.jl:151
 in lintstr(::String) at /home/galen/.julia/v0.5/Lint/src/Lint.jl:124
```

**Comment by TotalVerb** [comment]
This docstring syntax is not valid docstring syntax as far as I know. But this is nevertheless a bug that should be fixed.

**Comment by galenlynch** [comment]
You can at least access this type of field documentation in the REPL,
though I have been unable to find anything in the Julia documentation
explicitly mentioning it.

**Comment by TotalVerb** [comment]
@galenlynch How does one access this type of field documentation from the REPL?

**Comment by galenlynch** [comment]
You can access it by entering `type.field` in the help mode. For example:
``` Julia
julia> "A type doc string" type foo
       "a field doc string"
       bar::Int
       end

help?> foo.bar
  a field doc string
```

**Comment by TotalVerb** [comment]
Thanks, this is useful information. This issue is fixed by #179.

**Comment by galenlynch** [comment]
Thanks for fixing it!
